src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ2735SSMConverter.cpp

```diff
@@ -0,0 +1,164 @@
+#include "JsonToJ2735SSMConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    bool JsonToJ2735SSMConverter::parseJsonString(const string &consumedMsg, Json::Value &ssmDoc) const
+    {
+        const auto jsonLen = static_cast<int>(consumedMsg.length());
+        Json::CharReaderBuilder builder;
+        JSONCPP_STRING err;
+        const std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+        bool parseResult = reader->parse(consumedMsg.c_str(), consumedMsg.c_str() + jsonLen, &ssmDoc, &err);
+        if (!parseResult)
+        {
+            std::cout << "Parse error: " << err << endl;
+        }
+        return parseResult;
+    }
+
+    void JsonToJ2735SSMConverter::toJ2735SSM(const Json::Value &ssmDoc, std::shared_ptr<SignalStatusMessage> ssmPtr) const
+    {
+        if (!ssmDoc.isMember("SignalStatus"))
+        {
+            return;
```

Comment:
What happens if we return here without throwing exception. Does the calling method handle this error vase somehow.